apps/rush-lib/src/cli/utilities/LinkManager.ts

```diff
@@ -5,43 +5,31 @@ import * as colors from 'colors';
 import * as fsx from 'fs-extra';
 import * as os from 'os';
 import * as path from 'path';
-import * as semver from 'semver';
-import * as tar from 'tar';
-import readPackageTree = require('read-package-tree');
-import { JsonFile } from '@microsoft/node-core-library';
 
-import { RushConstants } from '../../RushConstants';
 import {
-  default as RushConfiguration,
-  IRushLinkJson
+  default as RushConfiguration
 } from '../../data/RushConfiguration';
-import RushConfigurationProject from '../../data/RushConfigurationProject';
 import Utilities from '../../utilities/Utilities';
 import { Stopwatch } from '../../utilities/Stopwatch';
-import Package, { IResolveOrCreateResult, PackageDependencyKind } from './Package';
-import PackageLookup from './PackageLookup';
+import Package from './Package';
 
-interface IQueueItem {
-  // A project from somewhere under "common/temp/node_modules"
-  commonPackage: Package;
-
-  // A symlinked virtual package that we will create somewhere under "this-project/node_modules"
-  localPackage: Package;
-
-  // If we encounter a dependency listed in cyclicDependencyProjects, this will be set to the root
-  // of the localPackage subtree where we will stop creating local links.
-  cyclicSubtreeRoot: Package | undefined;
-}
-
-enum SymlinkKind {
+export enum SymlinkKind {
   File,
   Directory
 }
 
-export default class LinkManager {
-  private _rushConfiguration: RushConfiguration;
+export default abstract class LinkManager {
+  protected _rushConfiguration: RushConfiguration;
+
+  public static getLinkManager(rushConfiguration: RushConfiguration): LinkManager {
+    return new (rushConfiguration.packageManager === 'pnpm'
```

Comment:
>rushConfiguration.packageManager === 'pnpm' [](start = 16, length = 43)

Over and over I keep seeing if-then statements that are considering PNPM vs NPM.  That's weird alongside an abstract baseclass with subclasses.

Reply:
I am just trying to provide a simple way to instantiate the correct subclass

---
In reply to: [151788372](https://github.com/Microsoft/web-build-tools/pull/427#discussion_r151788372) [](ancestors = 151788372)